gen/mailman

```diff
@@ -35,26 +35,25 @@ my $mailinglistsDirectory = "$DIRECTORY/mailinglists";
 mkdir $mailinglistsDirectory or die "Can't mkdir $mailinglistsDirectory: $!";
 
 
-my @membersAttributes;
-my @resourcesData = $data->getChildElements;
-foreach my $rData (@resourcesData) {
-    my %resourceAttributes  = attributesToHash $rData->getAttributes;
-    local *FILE;
+foreach my $resourceId ($data->getResourceIds()) {
 
-    my @membersData = $rData->getChildElements;
+	my $listName = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_NAME);
+	my $adminMail = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_MANAGER_MAIL);
 
-    my $listName = "$resourceAttributes{$A_RESOURCE_MAILING_LIST_NAME}";
-    my $adminMail = "$resourceAttributes{$A_RESOURCE_MAILING_LIST_MANAGER_MAIL}";
+	# Store the admin's mail
+	$mailinglistAdminMailStruc->{$listName} = $adminMail;
 
-    # Store the admin's mail
-    $mailinglistAdminMailStruc->{$listName} = $adminMail;
+	foreach my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
 
-    foreach my $memberAttributes (dataToAttributesHashes @membersData) {
-        #list only VALID members except if allowNonvalidUsers is true
-        next unless($memberAttributes->{$A_USER_STATUS} eq "VALID");
+		my $userStatus = $data->getMemberAttributeValue(member => $memberId, attrName => $A_USER_STATUS);
+		my $userName = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_NAME);
+		my $userMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_MAIL);
 
-        $mailinglistStruc->{$listName}->{$memberAttributes->{$A_USER_MAIL}}->{$A_USER_NAME} = $memberAttributes->{$A_USER_NAME};
-    }
+		#list only VALID members except if allowNonvalidUsers is true
```

Comment:
This comment is not true, since there is no checking of ``allowNonvalidUsers`` attribute.

Reply:
Fixed